url0e7d57473a_kyorohiro_HetimaUtil_tgz-pJ8-net_hetimatan_net_http_HttpServer3xxJ8/src/net/hetimatan/io/file/MarkableReaderHelper.java

```diff
@@ -0,0 +1,137 @@
+package net.hetimatan.io.file;
+
+
+import java.io.IOException;
+
+import net.hetimatan.util.http.HttpObject;
+import net.hetimatan.util.io.ByteArrayBuilder;
+
+public class MarkableReaderHelper {
+
+	
+	public static byte[] asciiAndGet(MarkableReader reader, byte[] forbit, int limit) throws IOException {
+		long start = reader.getFilePointer();
+		ascii(reader, forbit, limit);
+		long end = reader.getFilePointer();
+		if(start == end) {
+			return new byte[0];
+		} else {
+			byte[] ret = new byte[(int)(end-start)];
+			reader.seek(start);
+			reader.read(ret, 0, ret.length);
+			return ret;
+		}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 16:23, 51:58.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.